web/html/src/components/messages.tsx

```diff
@@ -123,29 +123,56 @@ export const fromServerMessage = (
   return msg;
 };
 
-function msg(severityIn: Severity, ...textIn: Array<React.ReactNode>) {
-  return textIn.map(function (txt) {
-    return { severity: severityIn, text: textIn };
-  });
+function msg(severityIn: Severity, textIn: React.ReactNode, listMultiple: boolean, header?: string) {
```

Comment:
I'm not sure if the addition of these flags makes sense to me. At the end of the day, the server is providing the error message, so the header etc should come from the server as well. With this setup you tightly couple the error handling logic between the backend and fronted.  

Instead, I think it would make more sense to simply allow the backend to send an error message with DOM if necessary. If we want to avoid multiple separate message boxes, the backend should not send multiple separate messages. What do you think @cbbayburt and @ncounter?

Reply:
> With this setup you tightly couple the error handling logic between the backend and frontend

I'm not sure to have fully understood what you meant, so please correct me if I got your point wrong.

The backend here has no knowledge of messages, it is just reporting the errors that have happened during the bootstrapping process in the response. 

In my opinion, it should be up to the frontend logic to decide how to show those errors, depending on the context. That's the reason why I added the additional parameters, to have an API to specify how to present multiple errors from the server. In my idea, the header is only for presentation purpose and has no meaning for the backend.

> At the end of the day, the server is providing the error message

Indeed, but ideally that should not be the case, in my opinion. But I don't think we have any kind of common error coding in place.

> I think it would make more sense to simply allow the backend to send an error message with DOM if necessary. If we want to avoid multiple separate message boxes, the backend should not send multiple separate messages

It seems to me that sending DOM or deciding the grouping on the backend side will lead to far more tight coupling, because implementing that would mean knowing in the backend how the UI works and looks like.

Reply:
Server-side DOM is not an option in this case, because in most places, the same errors are sent to media other than the web UI like API responses, logs, etc.

I think the separation of concerns is good here. It's basically: the server has multiple errors to display, and the frontend decides how to present them, with some formatting and additional explanatory text.

> In my idea, the header is only for presentation purpose and has no meaning for the backend.

I agree. It's not a part of the backend errors. It's mostly cosmetic and should be considered part of the page design.

I specifically wanted to have these flags (or some similar functionality) because the outcome of this PR is very neat, and we might consider using the same logic in more pages where multiple messages are possible (`products.tsx` in this PR, for example). The general problem we have with error reporting is that every page has its own implementation for such cases. With the introduction of reusable stuff like this, we'll move towards a more unified appearance.